noir-projects/noir-contracts/contracts/token_with_refunds_contract/src/main.nr

```diff
@@ -545,7 +545,14 @@ contract TokenWithRefunds {
         // 3. Deduct the funded amount from the user's balance - this is a maximum fee a user is willing to pay
         // (called fee limit in aztec spec). The difference between fee limit and the actual tx fee will be refunded 
         // to the user in the `complete_refund(...)` function.
-        storage.balances.sub(user, U128::from_integer(funded_amount)).emit(encode_and_encrypt_note_with_keys(&mut context, user_ovpk, user_ivpk, user));
+        let change = subtract_balance(
```

Comment:
I'm surprised this function is available here, I thought we had only added it to the token contract. Did we replicate those changes here?

At any point we should likely nuke `with_refunds` and merge it into the main token.

Reply:
Yes, I updated the token in a [PR down the stack](https://github.com/AztecProtocol/aztec-packages/pull/7969).